### ISSUE TITLE
save_as_image() as jpeg leads to entirely black image

### ISSUE THREAD

**Issue by RemPsyc** [davidgohel/flextable#381]
I noticed that per default, saving as `png` with `save_as_image` makes the table have a transparent background, so when checked on GitHub with dark mode, it is really difficult to read it:

![transparent background](https://user-images.githubusercontent.com/13123390/156968220-491f7842-5a9d-4ab7-a279-00a873e61080.png)

So I tried saving as jpeg instead, as those should have an actual non-transparent background. To my surprise, the backgrounds were instead all black! I've tried with the basic example provided in `?save_as_image`  with the same result. Demo:

``` r
library(flextable)
ft <- flextable( head( mtcars ) )
save_as_image(x = ft, path = "myimage.jpeg")
```
![myimage](https://user-images.githubusercontent.com/13123390/156967797-0f2cf8fe-e528-4a3d-855c-c55c82a082c0.jpg)

I've also tried using `webshot2` as well for the `png` (it doesn't support `jpeg`) but it didn't change the problematic transparent backgrounds.

### sessionInfo

```
sessionInfo()
R version 4.1.2 (2021-11-01)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19044)

Matrix products: default

locale:
[1] LC_COLLATE=English_Canada.1252  LC_CTYPE=English_Canada.1252   
[3] LC_MONETARY=English_Canada.1252 LC_NUMERIC=C                   
[5] LC_TIME=English_Canada.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] flextable_0.7.0

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.8        ps_1.6.0          digest_0.6.29     R6_2.5.1          jsonlite_1.8.0   
 [6] magrittr_2.0.2    evaluate_0.15     zip_2.2.0         rlang_1.0.1       gdtools_0.2.4    
[11] cli_3.2.0         uuid_1.0-3        data.table_1.14.2 callr_3.7.0       xml2_1.3.3       
[16] rmarkdown_2.12    webshot_0.5.2     tools_4.1.2       officer_0.4.1     processx_3.5.2   
[21] xfun_0.30         fastmap_1.1.0     compiler_4.1.2    systemfonts_1.0.4 base64enc_0.1-3  
[26] htmltools_0.5.2   knitr_1.37     

```

**Comment by davidgohel** [comment]
related to:

* #227
* https://ardata-fr.github.io/flextable-book/rendering.html#dependency-for-images

This is not something that I can fix.

* `jpeg` does not support transparency.
* webshot2 does not allow transparent background.
* webshot does allow transparent background.
* `png` does support transparency: see illustration below (made with your illustration)

<img width="1469" alt="Capture d’écran 2022-03-07 à 09 34 15" src="https://user-images.githubusercontent.com/4331618/156996505-d4c133f5-ca02-4572-aaf4-1f65a83ba4bb.png">

If the need is to display flextable in github dark mode, you can use a dark theme or create your own (see `flextable::theme_vader()`).

I am going to add details in the manual of `save_as_image()`.

**Comment by RemPsyc** [comment]
Thanks for your answer David! I really appreciate it (and I will never stop praising your package too 😅).

I have a few more questions though.

> `jpeg` does not support transparency.

1. Does that mean that flextables have a transparent background by default?
2. Does that mean there is no way or option to specify the background not to be transparent?
3. Does that mean that there is no point in saving any flextable as `jpeg`? 
4. If the latter is true, why does the help `?save_as_image` specify argument `path` as:
> path: image file to be created. It should end with .png, .pdf, <ins>or .jpeg.</ins>

To be clear, the example I provided outputs a fully black image even when not hosted on GitHub. Like literally opening the image in an image viewer, it is all black. Is that expected?

```
library(flextable)
ft <- flextable( head( mtcars ) )
save_as_image(x = ft, path = "myimage.jpeg")
```

**Comment by davidgohel** [comment]
To me, this is transparent:

![myimage](https://user-images.githubusercontent.com/4331618/157123554-d5def83b-9e32-4b0c-b942-e086df2ff840.png)


This is how it looks on my mac. 

<img width="891" alt="Capture d’écran 2022-03-07 à 22 50 30" src="https://user-images.githubusercontent.com/4331618/157123729-74ac68e4-513e-4fda-8215-a520b7d59d88.png">

I am done with screenshot now.


So your issue is about having a white background! Use the following code before starting any flextable production

```r
set_flextable_defaults(background.color = "white")
``` 

See https://ardata-fr.github.io/flextable-book/format.html#default-formatting-properties 

> `?flextable()` 
> Default formatting properties are automatically applied to every flextable you produce. You can change these default values with function [set_flextable_defaults()](http://127.0.0.1:29244/help/library/flextable/help/set_flextable_defaults).

Q:

1. yes, it's "transparent"
2. no, you could use `bg()` for example, but the suggested prop above is better, you will only need it once

> [...] Is that expected?

Yes! Software often add black or gray background. Watch the screenshot above and note I change this background color. My mac display it with gray bg.  

I don't know what to answer to Q3 and Q4. What do you want to know by asking these questions? If jpeg should be removed as option? That saving to jpeg has no sense?

**Comment by RemPsyc** [comment]
(I'm sorry I wasn't more clear earlier, in my previous code examples, I was saving as jpeg, not png, whereas I see in your example that you are saving as png. Your screenshots make sense to be transparent since they are png. )

And oh wow! Yes! You are right,

`set_flextable_defaults(background.color = "white")`

is exactly what I was looking for! When saving as png, it is now perfect (when saved as jpeg it does work as well and we can see the table clearly, but it also adds a large black border, though it doesn't matter since I prefer saving as png anyway). Hopefully, this thread and solution will be useful to future readers encountering this issue.

Thanks so much! You really did think of everything, it never ceases to amaze me :D Thanks so much for your patience with us all while you've put such a tremendous amount of work into perfecting this amazing package and documentation 🙏

PS: I have recently migrated my custom garage functions into a personal package, is there a proper way to let you know that your package is being integrated into other packages and show appreciation? Like dropping this info [here](https://github.com/RemPsyc/RemPsyc) as a sidenote, or is it better practice to open a new issue, email you, or message you through your website, or is there already another sort of system in place?
